sdcm/tester.py

```diff
@@ -834,22 +834,30 @@ def is_encrypt_keys_needed(self):
             'kmip_hosts' in append_scylla_yaml
         )
 
-    def prepare_kms_host(self) -> None:
-        version_supports_kms = (self.params.is_enterprise and
-                                ComparableScyllaVersion(self.params.scylla_version) >= '2023.1.3')
-        backend_support_kms = self.params.get('cluster_backend') in ('aws',)
-        kms_configured_in_sct = self.params.get('scylla_encryption_options')
-        test_uses_oracle = self.params.get("db_type") == "mixed_scylla"
-        should_enable_kms = (version_supports_kms and
-                             backend_support_kms and
-                             not kms_configured_in_sct and
-                             not test_uses_oracle and
-                             not self.params.get('enterprise_disable_kms'))
-
-        if should_enable_kms:
+    def prepare_kms_host(self) -> None:  # noqa: PLR0911
+        if self.params.get('cluster_backend') != 'aws':
+            logging.debug("Skip configuring AWS KMS, test is not running on AWS")
+            return
+        if self.params.get('enterprise_disable_kms'):
+            logging.debug("Skip configuring AWS KMS, `enterprise_disable_kms` is set in the config")
+            return
+        if self.params.get("db_type") == "mixed_scylla":
+            logging.debug("Skip configuring AWS KMS, test uses mixed scylla versions")
+            return
+        try:
+            scylla_version = ComparableScyllaVersion(self.params.scylla_version)
+            if not (self.params.is_enterprise and scylla_version >= '2023.1.3'):
```

Comment:
AFAIK, currently, the oldest ScyllaDB version in production is `2024.1.x`.
And it is very old already.
The `2023.1.3` is even more older.
Practically, it means we may skip this version check to increase probability to use AWS KMS when we intend to even if BYO image has no expected tags/labels.